rustfft fails to build

I've tried adding basic_dsp_matrix = "0.5.2" to Cargo.toml for correlation computation. That pulls in rustfft. Unfortunately, rustfft compilation fails:

Architecture is i686 (32bit). Any ideas?


```
   Compiling rustfft v2.0.0
error[E0369]: binary operation `*` cannot be applied to type `num_complex::Complex<T>`
  --> /data/pavel/.cargo/registry/src/github.com-1ecc6299db9ec823/rustfft-2.0.0/src/algorithm/mixed_radix.rs:95:24
   |
95 |             *element = *element * twiddle;
   |                        ^^^^^^^^^^^^^^^^^^
   |
   = note: an implementation of `std::ops::Mul` might be missing for `num_complex::Complex<T>`
...
error[E0369]: binary operation `*` cannot be applied to type `&num_complex::Complex<T>`
   --> /data/pavel/.cargo/registry/src/github.com-1ecc6299db9ec823/rustfft-2.0.0/src/algorithm/butterflies.rs:717:41
    |
717 |         *buffer.get_unchecked_mut(15) = buffer.get_unchecked(15) * twiddle9;
    |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: an implementation of `std::ops::Mul` might be missing for `&num_complex::Complex<T>`

error: aborting due to 73 previous errors

error: Could not compile `rustfft`.
```

I tried cargo +nightly run, but result is pretty similar:

```
error[E0369]: binary operation `*` cannot be applied to type `&num_complex::Complex<T>`
   --> /data/pavel/.cargo/registry/src/github.com-1ecc6299db9ec823/rustfft-2.0.0/src/algorithm/butterflies.rs:717:41
    |
717 |         *buffer.get_unchecked_mut(15) = buffer.get_unchecked(15) * twiddle9;
    |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: an implementation of `std::ops::Mul` might be missing for `&num_complex::Complex<T>`

error: aborting due to 73 previous errors

You've got a few errors: E0277, E0308, E0369, E0599, E0600
If you want more information on an error, try using "rustc --explain E0277"
error: Could not compile `rustfft`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,8 +10,8 @@ keywords = ["fft", "dft", "fourier", "signal"]
 license = "MIT OR Apache-2.0"
 
 [dependencies]
-num-complex = "0.1.36"
-num-traits = "0.1.37"
+num-complex = "0.1.42"
+num-traits = "0.2"
 
 [dev-dependencies]
-rand = "0.3.15"
+rand = "0.4"

